builder/vsphere/iso/step_create.go

```diff
@@ -105,10 +103,6 @@ func (c *CreateConfig) Prepare() []error {
 		c.GuestOSType = "otherGuest"
 	}
 
-	if c.Firmware != "" && c.Firmware != "bios" && c.Firmware != "efi" && c.Firmware != "efi-secure" {
-		errs = append(errs, fmt.Errorf("'firmware' must be 'bios', 'efi' or 'efi-secure'"))
-	}
```

Comment:
`CreateConfig` is prepared here:
https://github.com/hashicorp/packer/blob/9c1409dbba89daa33035dd4d7f56cb1605f28fb1/builder/vsphere/iso/config.go#L77

Meaning a configuration error could be detected by a `packer validate`.

Reply:
CreateConfig is not supposed to have `Firmware` anymore. I'm removing this field from it. 🤔

Reply:
Ah right and it is in `HardwareConfig` !! Sorry about that !